compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1554,29 +1520,24 @@ impl<'context> Elaborator<'context> {
 
             let generics = vecmap(&self.generics, |generic| generic.type_var.clone());
 
-            if let Err((prev_location, prev_file)) = self.interner.add_trait_implementation(
+            if let Err(prev_location) = self.interner.add_trait_implementation(
                 self_type.clone(),
                 trait_id,
                 trait_impl.impl_id.expect("impl_id should be set in define_function_metas"),
                 generics,
                 resolved_trait_impl,
             ) {
-                self.push_err(
-                    DefCollectorErrorKind::OverlappingImpl {
-                        typ: self_type.clone(),
-                        span: self_type_location.span,
-                    },
-                    self_type_location.file,
-                );
+                self.push_err(DefCollectorErrorKind::OverlappingImpl {
+                    typ: self_type.clone(),
+                    location: self_type_location,
+                });
 
+                // TODO: this comment isn't accurate anymore, maybe the two errors can be merged now?
```

Comment:
Please add a follow-up issue for this TODO

Reply:
I ended up fixing it in this PR.